lib/index.ts

```diff
@@ -85,10 +85,38 @@ class PWMetrics {
         const sheets = new Sheets(this.sheets, this.clientSecret);
         await sheets.appendResults(results.runs);
       }
+
+      if (this.resultHasExpectationErrors(results) && this.flags.expectations) {
+        throw new Error(messages.getMessage('HAS_EXPECTATION_ERRORS'));
+      }
     }
     return results;
   }
 
+  resultHasExpectationErrors(results: PWMetricsResults): boolean {
```

Comment:
I got your concern, but it's not valid in that case because we are working with `PWMetricsResults` which is related to `PWMetrics` file.